testing/tests/DevExpress.ui.widgets/fileManagerParts/toolbar.tests.js

```diff
@@ -200,4 +200,210 @@ QUnit.module("Toolbar", moduleConfig, () => {
+                {
+                    commandName: "create",
+                    locateInMenu: "always"
+                },
+                {
+                    commandName: "refresh",
+                    text: "Reinvigorate"
+                },
+                {
+                    commandName: "separator",
+                    location: "after"
+                },
+                {
+                    commandName: "viewMode",
+                    location: "before"
+                }]
+        });
+        this.clock.tick(400);
+
+        let $elements = this.wrapper.getGeneralToolbarElements();
+        assert.equal($elements.length, 5, "general toolbar has elements");
+
+        assert.ok($elements.eq(0).find(".dx-icon").hasClass(Consts.UPLOAD_ICON_CLASS), "show tree view button is rendered with new icon");
+        assert.ok($elements.eq(1).text().indexOf("Upload files") !== -1, "upload files button is rendered in new position");
+
+        const $dropDownMenuButton = this.wrapper.getDropDownMenuButton();
+        $dropDownMenuButton.trigger("dxclick");
+        this.clock.tick(400);
+        const $dropDownMenuItem = this.wrapper.getDropDownMenuItem(0);
+        assert.ok($($dropDownMenuItem).find(".dx-button-text").text().indexOf("New folder") !== -1, "create folder button is rendered in the dropDown menu");
+
+        assert.ok($elements.eq(2).val().indexOf("Details") !== -1, "view switcher is rendered in new location");
+        assert.ok($elements.eq(3).text().indexOf("Reinvigorate") !== -1, "refresh button is rendered with new text");
+
+
+        const $item = this.wrapper.findDetailsItem("File 1.txt");
+        $item.trigger("dxclick");
+        $item.trigger("click");
+        this.clock.tick(400);
+
+        let $toolbar = this.wrapper.getToolbar();
+        assert.ok($toolbar.hasClass(Consts.FILE_TOOLBAR_CLASS), "file toolbar displayed");
+    });
+
+    test("custom items render and modification", function(assert) {
+        let actionLog = [];
+        const log = function(logString) {
+            actionLog.push(logString);
+        };
+        const readLog = function(logStringIndex) {
+            if(!logStringIndex) {
+                logStringIndex = actionLog.length - 1;
+            }
+            return actionLog[logStringIndex];
+        };
+        const clearLog = function() {
+            actionLog = [];
+        };
+        function SayMeow() {
+            log("Meow!");
+        }
+        createFileManager(false);
+        this.clock.tick(400);
+
+        const fileManagerInstance = $("#fileManager").dxFileManager("instance");
+        fileManagerInstance.option("toolbar", {
+            generalItems: [
+                "showDirsPanel", "create", "upload", "refresh",
+                {
+                    commandName: "separator",
+                    location: "after"
+                },
+                "viewMode",
+                {
+                    commandName: "meow",
```

Comment:
It is better to check that a custom attribute (i.e. `customCommandName`) can be specified and can be accessed in `onClick()` function.